app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,151 @@
+  const [isColumnSelected, setIsColumnSelected] = useState<boolean>(false);
+  const [isDirectiveSelected, setIsDirectiveSelected] = useState<boolean>(false);
+  const [usageDirective, setUsageDirective] = useState<IUsageDirectives[]>([]);
+  const [directivesList, setDirectivesList] = useState<IDirectivesList[]>([]);
+  const directiveRef = useRef();
+  const classes = useStyles();
+
+  const handleDirectiveChange = (event: IOnRowClickValue | React.ChangeEvent<HTMLInputElement>) => {
+    if (!event.target.value) {
+      setIsDirectiveSelected(false);
+    }
+    setDirectiveInput(event.target.value);
+  };
+
+  useEffect(() => {
+    setUsageDirective(formatUsageDirectiveData(directiveInput, directivesList));
+  }, [directiveInput]);
+
+  const onEnterClick = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    {
+      const usageArraySplit =
+        usageDirective.length > 0 ? usageDirective[0]?.item?.usage.split(' ') : [];
+      const inputSplit = directiveInput.replace(/^\s+/g, '').split(' ');
+      if (event.key === 'Enter' && handlePasteDirective(directiveInput, directivesList)) {
+        onDirectiveInputHandler(directiveInput);
+      } else if (
+        event.key === 'Enter' &&
+        isColumnSelected &&
+        isDirectiveSelected &&
+        (usageArraySplit.length === inputSplit.length || inputSplit.length > usageArraySplit.length)
+      ) {
+        onDirectiveInputHandler(directiveInput);
+      }
+    }
+  };
+
+  return (
+    <>
+      {openDirectivePanel && (
+        <Box data-testid="directive-input-parent">
+          <AutoCompleteList
+            directiveInput={directiveInput}
+            onRowClick={(eventObject) => handleDirectiveChange(eventObject)}
+            setDirectivesList={setDirectivesList}
+            getDirectiveUsage={(activeResults: IUsageDirectives[], value) => {
+              setIsDirectiveSelected(value);
+              setUsageDirective(activeResults);
+            }}
+            onColumnSelected={() => {
+              setIsColumnSelected(true);
+            }}
+            isDirectiveSelected={isDirectiveSelected}
+            columnNamesList={columnNamesList}
+          />
+          <UsageDirectiveWrapper>
+            {Array.isArray(usageDirective) && usageDirective.length === 1 ? (
+              usageDirective.map((row: IUsageDirectives) => {
+                return <UsageDirective row={row} />;
+              })
+            ) : (
+              <></>
+            )}
+            <SearchBarWrapper>
+              <InputWrapper>
+                <label
+                  htmlFor="directive-input-search"
+                  data-testid="select-directive-input-label"
+                  className={classes.label}
+                >
+                  {T.translate('features.WranglerNewUI.GridPage.directivePanel.dollar')}
+                </label>
+                <input
+                  id="directive-input-search"
```

Comment:
i guess we dont need id
can u use data-testid instead

if id is indeed needed plz let me know the usecase

Reply:
can u plz add this in comments beside the code where id attribute is used
client will have a proper reason to believe here